cmds/record.c

```diff
@@ -1674,7 +1674,13 @@ static void check_perf_event(struct opts *opts)
 	bool found = false;
 	enum uftrace_pattern_type ptype = opts->patt_type;
 
-	has_perf_event = has_sched_event = !opts->no_event;
+	if (opts->no_event) {
+		has_perf_event = false;
+		has_sched_event = false;
+	}
```

Comment:
I think this part (and the change above) looks unnecessary.  Just adding the below is enough, no?

Reply:
> I think this part (and the change above) looks unnecessary.

Do you mean we don't have to have 2 if statements for `opts->no_event` and `opts->no_sched`?

> Just adding the below is enough, no?

Which below is enough?

Reply:
Your code right below here: `if (opts->no_sched) has_sched_event =false;`.  You can omit the brace too.

Reply:
> I think this part (and the change above) looks unnecessary. Just adding the below is enough, no?

1) Do you mean this?
```diff
-        if (opts->no_event) {
-                has_perf_event = false;
-                has_sched_event = false;
-        }
         if (opts->no_sched) {
                 has_sched_event = false;
         }
```
> Your code right below here: if (opts->no_sched) has_sched_event =false;. You can omit the brace too.

2) Do you mean this?
```diff
         if (opts->no_event) {
                 has_perf_event = false;
                 has_sched_event = false;
         }
-        if (opts->no_sched) {
+        if (opts->no_sched)
                 has_sched_event = false;
-        }
```

3) Otherwise this? (How do we switch off schedule event then?)
```diff
         if (opts->no_event) {
                 has_perf_event = false;
                 has_sched_event = false;
         }
-        if (opts->no_sched) {
-                has_sched_event = false;
-        }
```

Reply:
Yes, 1 + 2.  Also defaults back to false.

Reply:
Hmm.. I'm getting confused. Do you mean this?
```diff
diff --git a/cmds/record.c b/cmds/record.c
index c3fbc9cb..447d02f9 100644
--- a/cmds/record.c
+++ b/cmds/record.c
@@ -59,7 +59,7 @@ static pthread_mutex_t write_list_lock = PTHREAD_MUTEX_INITIALIZER;
 static bool buf_done;
 static int thread_ctl[2];

-static bool has_perf_event = true;
+static bool has_perf_event = false;
 static bool has_sched_event = true;
 static bool finish_received;

@@ -1674,13 +1674,8 @@ static void check_perf_event(struct opts *opts)
        bool found = false;
        enum uftrace_pattern_type ptype = opts->patt_type;

-       if (opts->no_event) {
-               has_perf_event = false;
-               has_sched_event = false;
-       }
-       if (opts->no_sched) {
+       if (opts->no_sched)
                has_sched_event = false;
-       }

        if (opts->event == NULL)
                return;
```

Reply:
Something like this?

```c
static bool has_perf_event;
static bool has_sched_event;

...

void check_perf_event(struct opts *opts)
{
   ...

   has_perf_event = has_sched_event = !opts->no_event;

   if (opts->no_sched)
      has_sched_event = false;
```

Reply:
Thanks. updated it again.